pennylane/pulse/transmon.py

```diff
@@ -338,7 +340,7 @@ def phase(phi0, t): return phi0 * (t - 0.5)
         H = qml.pulse.transmon_interaction(qubit_freqs, connections, g, wires=range(3))
 
         def amp(max_amp, t): return max_amp * jnp.sin(t) ** 2
-        def freq(fr, t): return fr
+        def freq(fr, t): return fr  # Enables frequency trainability
```

Comment:
I'm not a fan of the way frequency is defined here. Is casting a constant frequency is a tensor not enough? @lillian542 @dwierichs

Reply:
The frequency is handled this way in order to enable the trainability of the parameter. It is mentioned in the next paragraph so, in a first read, it does not make much sense. This is one of the things that got me wondering, which is why I added the comment. Maybe we could even add "see below" in the comment.

Reply:
We have the function `qml.pulse.constant` which exists solely for this purpose. In my opinion it would be nice to use that function here mainly for two reasons:
1. It makes clear that just passing a number is not the same, because "otherwise the PL devs would not have written a special function for this".
2. It will encourage users to do so as well.

Reply:
Thanks both of you for the context! I agree with David about changing the function to `qml.pulse.constant`.

Reply:
Since trainability is already a concept in PL, and it's more what @mudit2812 is referring to (trainable input format for parameters), maybe we could instead say something like `# create parametrized frequency with constant magnitude`. I'm not sure if that's the best. But something about making sure this is a parametrized value, rather than trainable per se.

It could also be that it should be in the text beforehand rather than a comment, if it was confusing. Like mention that we want the drive frequency to be an adjustable rather than fixed parameter in the Hamiltonian, so we define it as a callable with a constant output.

I agree with David about switching this out with `freq = qml.pulse.constant` instead.

Reply:
I just read this again and I don't like "frequency with constant magnitude". I take it back, don't say that. But maybe just `# create parametrized frequency` and some explanatory text above?

Reply:
I put all of this together updating the text and the code example. Let me know what you think 😄

Reply:
Looks good to me!